wagtailcache/cache.py

```diff
@@ -51,6 +52,20 @@ def _patch_header(response: HttpResponse, status: Status) -> None:
         response[wagtailcache_settings.WAGTAIL_CACHE_HEADER] = status.value
 
 
+def _clean_uri(uri: str):
+    """
+    remove Trailingslash , remove protocol and subdomain and decode URI.
+    Reasons:
+            - clean keyring
+            - problems with reverse Proxy
+            - problems with special characters e.g. Ü,Ä,ö,ß
+    """
+    expr = r"https://www\.|http://www\.|https://www|http://|https://|www\."
+    uri = re.sub(expr, "", uri).strip("/")
```

Comment:
I agree that it makes sense to remove the trailing slash....... however, I *think* that wagtail-cache will actually cache both URLs separately, as `/page` will be a 301 redirect, and `/page/` will be the content. But I'm not 100% sure if the slash redirect makes it into the cache, or if Django handles it before it hits the cache middleware. This will probably require manually inspecting the new keyring you made to find out what it does.

Reply:
https://github.com/coderedcorp/wagtail-cache/pull/22#discussion_r597785700